src/Steps/Release/PublishRelease.php

```diff
@@ -97,15 +97,16 @@ protected function releaseLibrary(OutputInterface $output, LibraryRelease $relea
         // Step 2: Tag and push this tag
         $this->publishTag($output, $releasePlanNode);
 
-        // Step 3: Create release in github
-        $this->createGitHubRelease($output, $releasePlanNode);
-
-        // Step 4: Rewrite composer.json to destabilise requirements
+        // Step 3: Rewrite composer.json to destabilise requirements
         ConstraintStabiliser::destabiliseConstraints($output, $releasePlanNode, false);
 
-        // Step 5: Push development branch to origin
+        // Step 4: Push development branch to origin
         $this->log($output, "Pushing branch <info>{$branch}</info>");
         $library->pushTo('origin');
+
+        // Step 5: Create release in github
+        // These needs to be last to ensure the branch has been pushed to github
+        $this->createGitHubRelease($output, $releasePlanNode);
```

Comment:
This step needs to be last because branches not pushed to github during the branch step, instead they are pushed during the publish step.  The github release api endpoint is fed the branch to get the changelog for the module.  It breaks if the branch does not exist on the remote